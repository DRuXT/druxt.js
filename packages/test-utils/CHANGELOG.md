# druxt-test-utils

## 0.2.15

### Patch Changes

- Updated dependencies [41cab3a0]
  - druxt@0.22.0

## 0.2.14

### Patch Changes

- Updated dependencies [e46a3290]
  - druxt@0.21.0

## 0.2.13

### Patch Changes

- Updated dependencies [49b67872]
  - druxt@0.20.0

## 0.2.12

### Patch Changes

- Updated dependencies [be21952]
  - druxt@0.19.0

## 0.2.11

### Patch Changes

- Updated dependencies [e6cf1fc]
  - druxt@0.18.3

## 0.2.10

### Patch Changes

- 540afca: Updated drupal-jsonapi-params to 2.0.0
- Updated dependencies [540afca]
  - druxt@0.18.2

## 0.2.9

### Patch Changes

- Updated dependencies [80164a1]
  - druxt@0.18.1

## 0.2.8

### Patch Changes

- Updated dependencies [e3d5238]
  - druxt@0.18.0

## 0.2.7

### Patch Changes

- Updated dependencies [b79701c]
  - druxt@0.17.0

## 0.2.6

### Patch Changes

- Updated dependencies [7b749bd]
  - druxt@0.16.0

## 0.2.5

### Patch Changes

- Updated dependencies [2ae1d6d]
  - druxt@0.15.0

## 0.2.4

### Patch Changes

- Updated dependencies [45bc0b9]
  - druxt@0.14.0

## 0.2.3

### Patch Changes

- Updated dependencies [77ab204]
  - druxt@0.13.0

## 0.2.2

### Patch Changes

- Updated dependencies [4504a2f]
  - druxt@0.12.0

## 0.2.1

### Patch Changes

- Updated dependencies [75ff8a9]
  - druxt@0.11.0

## 0.2.0

### Patch Changes

- Updated dependencies [715e5ef]
  - druxt@0.10.0

## 0.1.0

### Patch Changes

- Updated dependencies [21170fb]
  - druxt@0.9.0
