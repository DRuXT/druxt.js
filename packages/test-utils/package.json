{
  "name": "druxt-test-utils",
  "version": "0.1.0",
  "description": "Test utilities for Druxt development.",
  "keywords": [
    "druxt",
    "jsonapi",
    "test"
  ],
  "exports": {
    ".": {
      "require": "./dist/druxt-test-utils.ssr.js",
      "import": "./dist/druxt-test-utils.esm.js"
    }
  },
  "main": "dist/druxt-test-utils.ssr.js",
  "module": "dist/druxt-test-utils.esm.js",
  "dependencies": {
    "drupal-jsonapi-params": "^1.2.2",
    "jest-mock-axios": "^4.4.0"
  },
  "peerDependencies": {
    "druxt": "*"
  }
}
