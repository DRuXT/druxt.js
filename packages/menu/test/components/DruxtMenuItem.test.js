import { config, createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'

import { DruxtMenu, DruxtMenuStore } from '../..'
import DruxtMenuComponent from '../../src/components/DruxtMenu.vue'
import DruxtMenuItemComponent from '../../src/components/DruxtMenuItem.vue'

jest.mock('axios')

const baseUrl = 'https://demo-api.druxtjs.org'

const propsData = {
  item: {
    entity: {
      attributes: {
        title: 'Parent',
        link: { uri: 'internal:/parent' },
        route: { name: 'test' }
      }
    },
    children: [
      {
        entity: {
          attributes: {
            title: 'Child - Internal',
            link: { uri: 'entity:node/1' },
            route: { name: 'test-child' }
          }
        },
        children: []
      },
      {
        entity: {
          attributes: {
            title: 'Child - External',
            link: { uri: 'entity:https://druxtjs.org' },
            route: { name: '' }
          }
        },
        children: []
      },
    ]
  }
}

const stubs = ['nuxt-link']

// Setup local vue instance.
const localVue = createLocalVue()
localVue.use(Vuex)

const mountComponent = options => {
  config.mocks.$route = {
    path: '/'
  }

  return shallowMount(DruxtMenuItemComponent, {
    localVue,
    propsData,
    parentComponent: DruxtMenuComponent,
    stubs,
    ...options
  })
}

let store

describe('DruxtMenuItem', () => {
  beforeEach(() => {
    // Setup vuex store.
    store = new Vuex.Store()
    store.$druxtMenu = new DruxtMenu(baseUrl, {})
    DruxtMenuStore({ store })
    config.mocks.$store = store
  })

  test('default', () => {
    const wrapper = mountComponent()
    // Ensure we get sane default HTML.
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.vm.to).toStrictEqual({ path: '/parent' })
  })

  test('unwrapped', () => {
    const wrapper = mountComponent({ parentComponent: null })
    // Ensure unwrapped instance of `<druxt-menu-item />` doesn't render.
    expect(wrapper.html()).toBe('')
  })

  test('external', () => {
    const wrapper = mountComponent({
      propsData: {
        item: {
          entity: {
            attributes: {
              title: 'External',
              url: 'https://druxtjs.org',
              route: { name: '' }
            }
          }
        }
      }
    })

    // Ensure we get sane default HTML.
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.vm.to).toStrictEqual(false)
  })
})
