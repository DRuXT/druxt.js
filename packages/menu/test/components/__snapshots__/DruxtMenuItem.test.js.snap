// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DruxtMenuItem default 1`] = `
"<li class=\\"\\">
  <nuxt-link-stub>Parent</nuxt-link-stub>
  <ul>
    <druxt-menu-item-stub item=\\"[object Object]\\"></druxt-menu-item-stub>
  </ul>
</li>"
`;
